From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Damien Caliste <dcaliste@free.fr>
Date: Tue, 8 Oct 2024 16:59:57 +0200
Subject: [PATCH] Fallback to sso credential plugin.

By default, if auth/plugin key is absent, the PlainCredentials are used
(i.e. internal storage). For Sailfish OS, fallback to the SSO plugin.
---
 .../qmfmessageserver/qmailcredentials.cpp         | 15 +++++++--------
 1 file changed, 7 insertions(+), 8 deletions(-)

diff --git a/src/libraries/qmfmessageserver/qmailcredentials.cpp b/src/libraries/qmfmessageserver/qmailcredentials.cpp
index 8e324cc5..b0bd01b3 100644
--- a/src/libraries/qmfmessageserver/qmailcredentials.cpp
+++ b/src/libraries/qmfmessageserver/qmailcredentials.cpp
@@ -187,14 +187,13 @@ QMailCredentialsInterface *QMailCredentialsFactory::getCredentialsHandlerForAcco
     const QMailAccountConfiguration::ServiceConfiguration &auth = config.serviceConfiguration(QLatin1String("auth"));
     if (auth.id().isValid()) {
         const QString plugin = auth.value(QLatin1String("plugin"));
-        if (!plugin.isEmpty()) {
-            credentials = QMailCredentialsFactory::createCredentialsHandler(plugin, parent);
-            if (!credentials) {
-                qCWarning(lcMessaging) << "Credential plugin" << plugin
-                                       << "is not available for account id: "
-                                       << config.id()
-                                       << ", account configuration will be used";
-            }
+        credentials = QMailCredentialsFactory::createCredentialsHandler
+            (plugin.isEmpty() ? QLatin1String("sso") : plugin, parent);
+        if (!credentials) {
+            qCWarning(lcMessaging) << "Credential plugin" << plugin
+                                   << "is not available for account id: "
+                                   << config.id()
+                                   << ", account configuration will be used";
         }
     }
     if (!credentials) {
